posydon/unit_tests/utils/test_common_functions.py

```diff
@@ -0,0 +1,2065 @@
+            BinaryStar.star_1.state = CO
+            assert totest.eddington_limit(BinaryStar) == r
+        BinaryStar.star_1.state = None
+        # examples: 1.2Msun accretor is star2, while donor has X_surf=0.1
+        BinaryStar.star_2.mass = 1.2
+        for CO, r in zip(['WD', 'NS', 'BH'],\
+            [(approx(5.89502616630e-17, abs=6e-29),\
+              approx(8.16917025429e+7, abs=6e-5)),\
+             (approx(3.39586943808e-8, abs=6e-20),\
+              approx(0.14181190792, abs=6e-12)),\
+             (approx(8.42046955292e-8, abs=6e-20),\
+              approx(5.71909584179e-2, abs=6e-14))]):
+            BinaryStar.star_2.state = CO
+            BinaryStar.star_1.surface_h1 = 0.1
+            assert totest.eddington_limit(BinaryStar) == r
+        # examples: 1.3Msun accretor is star1 with history
+        BinaryStar.star_1.state = 'BH'
+        BinaryStar.star_1.mass_history = totest.np.array([1.3, 1.3])
+        BinaryStar.star_1.state_history = totest.np.array([None, None])
+        with raises(ValueError, match='COtype must be "BH", "NS", or "WD"'):
+            totest.eddington_limit(BinaryStar, idx=0)
+        for CO, r in zip(['WD', 'NS', 'BH'],\
+            [(approx(3.68044499210e-17, abs=6e-29),\
+              approx(9.08923688740e+7, abs=6e-5)),\
+             (approx(2.17747384546e-8, abs=6e-20),\
+              approx(0.15362956691, abs=6e-12)),\
+             (approx(5.84925262833e-8, abs=6e-20),\
+              approx(5.71909584179e-2, abs=6e-14))]):
+            BinaryStar.star_1.state_history = totest.np.array([None, CO])
+            assert totest.eddington_limit(BinaryStar, idx=0) == r
+
+    def test_beaming(self, BinaryStar):
+        # missing argument
+        with raises(TypeError, match="missing 1 required positional"+\
+                    " argument: 'binary'"):
+            totest.beaming()
+        # bad input
+        with raises(ValueError, match="Eddington limit is being calculated"+\
+                    " for a non-CO"):
+            totest.beaming(BinaryStar)
+        # examples: 1Msun accretor is star1 with no mass-transfer rate
+        BinaryStar.star_1.mass = 1.0
+        for CO, r in zip(['WD', 'NS', 'BH'],\
+            [(totest.np.nan, 1),\
+             (totest.np.nan, 1),\
+             (totest.np.nan, 1)]):
+            BinaryStar.star_1.state = CO
+            assert totest.beaming(BinaryStar) == r
+        # examples: 1.2Msun accretor is star1
+        BinaryStar.star_1.mass = 1.2
+        BinaryStar.lg_mtransfer_rate = -7.5
+        for CO, r in zip(['WD', 'NS', 'BH'],\
+            [(approx(7.80787388850, abs=6e-12),\
+              approx(1.04303350643e-16, abs=6e-28)),\
+             (approx(2.98994840792e-8, abs=6e-20), 1),\
+             (approx(3.16227766017e-8, abs=6e-20), 1)]):
+            BinaryStar.star_1.state = CO
+            assert totest.beaming(BinaryStar) == r
+
+    def test_bondi_hoyle(self, BinaryStar, monkeypatch):
```

Comment:
Yes, it is numerically valid input, but non-physical to have a compact object to be the donor and a non-compact object to be the accretor.
Additionally, it will only cause the `eddington_limit` function to fail, which is tested in the test of that function.